회원, 장소 - 예외처리

hiYong94 opened the pull request.

hiYong94 committed:
proejct upload

hiYong94 committed:
project upload

hiYong94 committed:
feat: test를 위한 배포

hiYong94 committed:
code refactoring

hiYong94 committed:
errorhandler create

hiYong94 committed:
html test

hiYong94 committed:
function parameter update

hiYong94 committed:
code test & update

hiYong94 committed:
code confirm update, value check and error handling

hiYong94 committed:
conflicts merge

hiYong94 committed:
오타수정

hiYong94 committed:
add value check

innovator777 commented:
그냥....error.js 를 지워내고 각 function에서 response.json()하는 형태로 가는게 좋아보이네요

innovator777 commented on service/placeService.js:
```diff
@@ -0,0 +1,112 @@
+            error.status = 500
+            error.message = str2json.convert({ "status": 500, "message": error.message})
+            next(error)
+        }
```
만약 update에 필요한 데이터가 안들어 오거나 이상하게 들어왔을 때도  500 error를 response로 반환할것 같은데...
그렇게 된다면 정상적인 error code를 반환한 상황인 건가요?

hiYong94 replied:
> 만약 update에 필요한 데이터가 안들어 오거나 이상하게 들어왔을 때도 500 error를 response로 반환할것 같은데...
> 그렇게 된다면 정상적인 error code를 반환한 상황인 건가요?

아직 제대로 수정이 안되있어서 그런것 같습니다! 마저 수정 후 정확한 코드로 다시 커밋하겠습니다.

innovator777 commented on service/placeService.js:
```diff
@@ -0,0 +1,112 @@
+        if (error) {
+            error.status = 500
+            error.message = str2json.convert({ "status": 500, "message": error.message})
+            next(error)
```
위의 리뷰와 같습니다.

전체적으로 다 마찬가지인 것 같네요

hiYong94 commented:
> 그냥....error.js 를 지워내고 각 function에서 response.json()하는 형태로 가는게 좋아보이네요

사실 error.js로 함수화해서 쓰고싶은데 function의 error와 router의 error에 대한 구분을 정확히 못하고있어서 error.js 내부에 한번에 사용하고 있지 못하고있습니다.. 좀더 확인해보고 함수화 가능하며 함수화로 통일하거나 아니면 response.json()의 형태로 통일하겠습니다!

hiYong94 committed:
server update

hiYong94 committed:
Code refactoring and Add function

* add costomModules/ error handle and value check function
* add image file upload
* code refactoring user/, place/
* do not use error.js
* update app.js

hiYong94 committed:
Modify query

* Add parameters latitude, longitude

hiYong94 committed:
Modify insert keyword

* insert keyword array modify

hiYong94 committed:
Add search function

* search for a place by name on the map

hiYong94 committed:
Fix sql query

hiYong94 committed:
Add function keyword

* add keyword API get function

hiYong94 committed:
Fix sql query

* search API sql query modify

hiYong94 committed:
Fix sql query

* search API sql query modify

hiYong94 committed:
image upload

hiYong94 committed:
Fix response json

* Fix response json data test

hiYong94 committed:
Fix response json

* response json data test

hiYong94 committed:
Fix response json

* response json data test

hiYong94 committed:
Delete test images

hiYong94 committed:
Fix keyword service and dao

* modified keyword select all

Add place_has_keyword transaction query

* add function place_has_keyword table transaction insert query

hiYong94 committed:
Fix delete user

* modified delete function return error

hiYong94 committed:
Place create function add & modify

* Add images upload router/place/place.js
* Add images upload and keyword refactoring service/placeService.js
  - Refactor keywordName Array type
  - Add images upload function
* Fix plade readOnePlace response data
  - response json list to response json single data
* Refactor place insert query transaction
  - Refactor place insert query transaction
  - Refactor & Add place insert query transaction and function module
    > place, place_images, place_has_keyword table insert query

hiYong94 committed:
Delete comment

hiYong94 committed:
Fix keyword query request

hiYong94 committed:
Add test images folder upload

hiYong94 committed:
Add place insert transaction function

* Add place_images table insert query

hiYong94 committed:
Delete semicolon

hiYong94 committed:
Add Test Api router

hiYong94 committed:
Add search sql query for map search by place name
Fix map search RestApi GET

* add place by name sql query
* modfiy map search get api

hiYong94 committed:
Add RestAPI search

* add search place by name
* add search address by name
* add search keyword by name

hiYong94 committed:
Add RestAPI Post reivew function

hiYong94 committed:
Add review images

hiYong94 committed:
Add membership function

hiYong94 committed:
Fix map search

hiYong94 committed:
Add image uploads

hiYong94 committed:
Add review function

* add review post and get function

hiYong94 committed:
Fix function parameter

hiYong94 committed:
Add images null check
